src/lib/x509/x509_obj.h

```diff
@@ -69,9 +64,11 @@ class BOTAN_PUBLIC_API(2,0) X509_Object : public ASN1_Object
       * Check the signature on this data
       * @param key the public key purportedly used to sign this data
       * @return status of the signature - OK if verified or otherwise an indicator of
-      *         the problem preventing verification.
+      *         the problem preventing verification, along with the hash function that
+      *         was used, for further policy checks. The second parameter is empty
+      *         unless the validation was sucessful.
       */
-      Certificate_Status_Code verify_signature(const Public_Key& key) const;
+      std::pair<Certificate_Status_Code, std::string> verify_signature(const Public_Key& key) const;
```

Comment:
I personally am not a fan of `std::pair` in public APIs. Did you consider adding a small helper struct for it so the parts are properly named then accessed?

Reply:
Might not be a bad idea. Though this function while nominally public mostly exists to help implement other functionality so I'm not sure it matters that much in this particular case.